### PR TITLE
Tags : introduce set_is_favorite and set_is_low_priority

### PR DESCRIPTION
**Pull request by ganfra**
Handles "Expose a way to update tags in the bindings" from https://github.com/matrix-org/matrix-rust-sdk/issues/3005

### CODE REVIEW

File: crates/matrix-sdk/src/room/mod.rs [hunk 1]
```diff
@@ -937,6 +937,20 @@ impl Room {
         self.client.send(request, None).await
     }
 
+    /// Update the tags from the room by calling set_tag or remove_tag method if
+    /// needed.
+    pub async fn update_notable_tags(&self, notable_tags: RoomNotableTags) -> Result<()> {
```
Comment (ganfra) [inline crates/matrix-sdk/src/room/mod.rs:942]: I'm not sure this is best to do like this for the future, or offer method for each tags like
`set_is_favorite(bool)`
`set_is_low_priority(bool)`

**Review by andybalaam — CHANGES_REQUESTED** [review]
This is great, thank you. Please could you split the test up to make it easier to understand?

I would recommend:

1. Several little helper methods e.g. `mock_favourite` to do the mocking stuff, `do_sync` (or a better name!) to do the sync stuff
2. Break it into several tests with names like `updating_notable_tags_causes_new_favourite_to_be_produced` (or something better)

I really like my main test methods to be easily readable as "set up, do something, assert something" ("Given", "When", "Then" according to some people). At the moment, it takes a lot of reading to see what the test is really testing.

Thank you!

File: crates/matrix-sdk/src/room/mod.rs (hunk 1 above)
Comment (andybalaam) [reply crates/matrix-sdk/src/room/mod.rs:942]: Me neither. @bnjbvr do you have an opinion?

**Comment by ganfra** [comment]
> This is great, thank you. Please could you split the test up to make it easier to understand?
> 
> I would recommend:
> 
> 1. Several little helper methods e.g. `mock_favourite` to do the mocking stuff, `do_sync` (or a better name!) to do the sync stuff
> 2. Break it into several tests with names like `updating_notable_tags_causes_new_favourite_to_be_produced` (or something better)
> 
> I really like my main test methods to be easily readable as "set up, do something, assert something" ("Given", "When", "Then" according to some people). At the moment, it takes a lot of reading to see what the test is really testing.
> 
> Thank you!

Honestly I wrote the tests in the same way other tests have been written, but I agree they are not easy to read..

File: crates/matrix-sdk/src/room/mod.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk/src/room/mod.rs:942]: I don't find it likely to change multiple ones at the same time, this shouldn't happen quite often so performance isn't critical, and having a single method for each notable tag makes for a better API IMO, so I'd in favor of having `set_is_...` methods :+1:

Comment (ganfra) [reply crates/matrix-sdk/src/room/mod.rs:942]: Do you think it's worth checking the local value tags before sending the request? 
I guess we can have some race condition (especially with [this issue](https://github.com/matrix-org/sliding-sync/issues/189)) which could prevent the request to happen when it should.

**Comment by ganfra** [comment]
@andybalaam ok so, I changed the method `update_notable_tags` with `set_is_favorite` and `set_is_low_priority`.
I also added the `is_low_priority` in `RoomNotableTags`.
And I've split all the tests so it's more readable.

Let me know what you think!

**Review by andybalaam — APPROVED** [review]
The tests look much better - I was much more able to understand what you were doing - thank you!

The code looks good to me, but I will want a nod from @bnjbvr before I merge, to check this is the right approach overall.

File: crates/matrix-sdk-base/src/rooms/normal.rs [hunk 2]
```diff
@@ -655,11 +655,16 @@ impl Room {
 
     /// Returns the current RoomNotableTags and subscribe to changes.
```
[the remark this answers is not in the document]
Comment (ganfra) [reply crates/matrix-sdk-base/src/rooms/normal.rs:656]: @Hywan I let you handle those changes on your own. There is tons of places where [` `] notation is not used (almost everywhere actually).

[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk-base/src/rooms/normal.rs:656]: This is your code. Can you fix it please? I'll handle the others, but please handle it in your code.

**Review by bnjbvr — APPROVED** [review]

**Review by Hywan — APPROVED** [review]

**Pull request merged**